hbase_storage/hbase1.1.1/src/main/java/com/splicemachine/storage/ClientPartition.java

```diff
@@ -286,9 +286,14 @@ public Table unwrapDelegate(){
     private List<HRegionLocation> getAllRegionLocations(boolean refresh) throws IOException {
         if (refresh)
            ((HConnection) connection).clearRegionCache(tableName);
-        try(RegionLocator regionLocator=connection.getRegionLocator(tableName)){
-            return regionLocator.getAllRegionLocations();
+        while (true) {
+            try (RegionLocator regionLocator = connection.getRegionLocator(tableName)) {
+                return regionLocator.getAllRegionLocations();
+            } catch (Exception e) {
+                SpliceLogUtils.error(LOG,String.format("getAllRegionLocations for tablename=%s failing",tableName),e);
+            }
         }
```

Comment:
why not wait a while and retry?

Reply:
In addition to the wait, I would suggest having a maximum number of retries. Having an unconditional loop here seems bound to bite us in the ass at some point.